tests/usermetas.test.js

```diff
@@ -1,21 +1,108 @@
 'use strict';
 
+const {
+	assert
+} = require('chai');
 const Helper = require('../server/users/helpers');
+const router = require('../server/users/index');
 
-let userId = 'test1';
 
 describe('usermetas', async () => {
-    describe('#dynamicMetadata()', () => {
-      it('returns user metas based on a test user', async () => {
-        const helper = td.object(Helper);  
-  
-        td.when(helper.dynamicMetadata(userId, 'location,locale,auth', td.callback))
-          .thenCallback(null, 'test');
-  
-        
-      });
-   
-   
-    });
-  });
-  
\ No newline at end of file
+	describe('#dynamicMetadata()', () => {
+		it('returns user metas based on a test user', (done) => {
+			const userId = 'test1';
+			const metadataList = ['usermeta1', 'usermeta2'];
+			const helper = td.object(Helper);
+			const response = [{
+				id: "usermeta1",
+				value: ""
+			}, {
+				id: "usermeta2",
+				value: ""
+			}];
+
+			td.when(helper.dynamicMetadata(userId, metadataList.join(','), td.callback))
+				.thenCallback(null, response);
+
+			const res = td.object(['json']);
+			const req = {
+				params: {
+					username: userId
+				}
+			};
+			const next = function(err) {
+				try {
+					assert(err == undefined, 'there should not be an error');
+					td.verify(res.json(response));
+					done();
+				} catch (e) {
+					done(e);
+				}
+			};
+			router.getUsermetaMiddleware(helper, metadataList.join(','))(req, res, next);
+		});
+
+		it('checking metadatalist emtpy', (done) => {
+			const userId = 'test1';
+			const metadataList = [];
+			const helper = td.object(Helper);
+			const response = [{
+				id: "",
+				value: ""
+			}];
+
+			td.when(helper.dynamicMetadata(userId, metadataList.join(','), td.callback))
+				.thenCallback(null, response);
+
+			const res = td.object(['json']);
+			const req = {
+				params: {
+					username: userId
+				}
+			};
+			const next = function(err) {
+				try {
+					assert(err == undefined, 'there should not be an error');
+					td.verify(res.json(response));
+					done();
+				} catch (e) {
+					done(e);
+				}
+			};
+			router.getUsermetaMiddleware(helper, metadataList.join(','))(req, res, next);
+		});
+
+    it('Updating metadata', (done) => {
+			const userId = 'test1'; 
+      const metaKey = 'usermeta1';
+      const metaVal = 'from-tests';
+			const helper = td.object(Helper);
+			const response = {};
+
+			td.when(helper.updateDynamicUserMeta(userId, metaKey, metaVal, td.callback))
+				.thenCallback(null, response);
+
+			const res = td.object(['json']);
+			const req = {
+				params: {
+					username: userId, 
+				},
+        body: {
+          id: metaKey,
+          value: metaVal
+        }
+			};
+			const next = function(err) {
+				try {
+					assert(err == undefined, 'there should not be an error');
+					td.verify(res.json(response));
```

Comment:
I believe here the important point in the test is to verify that `helper.updateDynamicUserMeta` has been called with the correct parameters. You could do:

`1` instruct updateDynamicUserMeta to always return the response, regardless of arguments.
```js
td.when(helper.updateDynamicUserMeta(td.matchers.anything(), td.matchers.anything(), td.matchers.anything(), td.callback))
  .thenCallback(null, response);
```

`2`, in the next function:
```js
td.verify(helper.updateDynamicUserMeta(userId, metaKey, metaVal, td.callback));
```

Why is this better?

Currently, in case case `updateDynamicUserMeta` wasn't called with the correct arguments, this will fail with a timeout. With the proposed change, the test will fail but telling us that `updateDynamicUserMeta` wasn't called with the correct arguments.

Also, it makes it clearer what is the important feature of this middleware we are testing: it should update the usermeta to the correct value.